libgadget/forcetree.c

```diff
@@ -357,6 +357,7 @@ create_new_node_layer(int firstparent, int p_toplace,
                     oldsuns[0], P[oldsuns[0]].Pos[0], P[oldsuns[0]].Pos[1], P[oldsuns[0]].Pos[2], P[oldsuns[0]].Type, P[oldsuns[0]].ID,
                     oldsuns[1], P[oldsuns[1]].Pos[0], P[oldsuns[1]].Pos[1], P[oldsuns[1]].Pos[2], P[oldsuns[1]].Type, P[oldsuns[1]].ID
                 );
+                nc->nnext_thread = tb.lastnode + 10 * NODECACHE_SIZE;
```

Comment:
Or perhaps you are using nc->nnext_thread to signal loop termination?

Reply:
Right